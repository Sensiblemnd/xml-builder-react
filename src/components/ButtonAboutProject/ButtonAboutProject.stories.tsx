import * as React from "react";

import { storiesOf } from "@storybook/react";
import ButtonAboutProject from "./ButtonAboutProject";

storiesOf("Button | About Project", module).add("Default", () => (
  <React.Fragment>
    Click me!
    <ButtonAboutProject
      version="1.0.0"
      productName="Mi producto"
      githubSourceCodeURL="https://github.com/project-openubl/xml-builder"
      githubIssuesURL="https://github.com/project-openubl/xml-builder/issues"
      documentationURL="https://project-openubl.gitbook.io/xml-builder/"
      swaggerApiURL="https://app.swaggerhub.com/organizations/project-openubl"
    />
  </React.Fragment>
));
