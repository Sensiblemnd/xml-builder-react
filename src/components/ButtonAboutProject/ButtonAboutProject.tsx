import React from "react";
import {
  AboutModal,
  TextContent,
  TextList,
  TextListItem,
  Button,
  ButtonVariant
} from "@patternfly/react-core";
import { HelpIcon } from "@patternfly/react-icons";
import brandImg from "../../img/logo.png";

export interface Props {
  version: string;
  productName: string;
  githubSourceCodeURL: string;
  githubIssuesURL: string;
  documentationURL: string;
  swaggerApiURL: string;
}

export interface State {
  isOpen: boolean;
}

class ButtonAboutProject extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      isOpen: false
    };
  }

  toggleButton = () => {
    this.setState(({ isOpen }) => ({
      isOpen: !isOpen
    }));
  };

  render() {
    const {
      version,
      productName,
      githubSourceCodeURL,
      githubIssuesURL,
      documentationURL,
      swaggerApiURL
    } = this.props;

    const { isOpen } = this.state;

    return (
      <React.Fragment>
        <Button
          id="aboutButton"
          aria-label="About button"
          variant={ButtonVariant.plain}
          onClick={this.toggleButton}
        >
          <HelpIcon />
        </Button>
        <AboutModal
          isOpen={isOpen}
          onClose={this.toggleButton}
          trademark="COPYRIGHT Â© 2020. PROJECT OPENUBL"
          brandImageSrc={brandImg}
          brandImageAlt="Logo"
          productName={productName}
        >
          <TextContent>
            <TextList component="dl">
              <TextListItem component="dt">Version</TextListItem>
              <TextListItem component="dd">{version}</TextListItem>
              <TextListItem component="dt">Source code</TextListItem>
              <TextListItem component="dd">
                <a href={githubSourceCodeURL}>Github</a>
              </TextListItem>
              <TextListItem component="dt">Report issues</TextListItem>
              <TextListItem component="dd">
                <a href={githubIssuesURL}>Github issues</a>
              </TextListItem>
              <TextListItem component="dt">Documentation</TextListItem>
              <TextListItem component="dd">
                <a href={documentationURL}>Documentation</a>
              </TextListItem>
              <TextListItem component="dt">Videos</TextListItem>
              <TextListItem component="dd">
                <a href="https://www.youtube.com/channel/UChq3xxjyDgjcU346rp0bbtA/">
                  Youtube
                </a>
              </TextListItem>
              <TextListItem component="dt">Rest API documentation</TextListItem>
              <TextListItem component="dd">
                <a href={swaggerApiURL}>Swaggerhub</a>
              </TextListItem>
              <TextListItem component="dt">Licence</TextListItem>
              <TextListItem component="dd">
                <a href="https://github.com/project-openubl/xml-builder/blob/master/LICENSE">
                  Eclipse Public License - v 2.0
                </a>
              </TextListItem>
            </TextList>
          </TextContent>
        </AboutModal>
      </React.Fragment>
    );
  }
}

export default ButtonAboutProject;
