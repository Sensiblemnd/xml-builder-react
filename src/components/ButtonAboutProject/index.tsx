import ButtonAboutProject from "./ButtonAboutProject";

export default ButtonAboutProject;
