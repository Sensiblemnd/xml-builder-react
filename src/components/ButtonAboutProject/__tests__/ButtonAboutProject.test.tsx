import * as React from "react";
import { shallow } from "enzyme";
import { ButtonAboutProject } from "../ButtonAboutProject";
import brandImageSrc from "../../../styles/assets/images/logo.png";

it("Renders without crashing", () => {
  const wrapper = shallow(
    <ButtonAboutProject
      version="1.0.0"
      productName="Mi producto"
      githubSourceCodeURL="https://github.com/project-openubl/xml-builder"
      githubIssuesURL="https://github.com/project-openubl/xml-builder/issues"
      documentationURL="https://project-openubl.gitbook.io/xml-builder/"
      swaggerApiURL="https://app.swaggerhub.com/organizations/project-openubl"
      brandImageSrc={brandImageSrc}
    />
  );

  expect(wrapper).toMatchSnapshot();
});
