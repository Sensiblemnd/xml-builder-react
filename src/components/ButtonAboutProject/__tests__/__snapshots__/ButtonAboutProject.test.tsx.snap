// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing 1`] = `
<Fragment>
  <Component
    aria-label="About button"
    id="aboutButton"
    onClick={[Function]}
    variant="plain"
  >
    <HelpIcon
      color="currentColor"
      noVerticalAlign={false}
      size="sm"
      title={null}
    />
  </Component>
  <AboutModal
    appendTo={null}
    backgroundImageSrc=""
    brandImageAlt="Logo"
    brandImageSrc="logo.png"
    className=""
    isOpen={false}
    noAboutModalBoxContentContainer={false}
    onClose={[Function]}
    productName="Mi producto"
    trademark="COPYRIGHT Â© 2020. PROJECT OPENUBL"
  >
    <TextContent>
      <TextList
        component="dl"
      >
        <TextListItem
          component="dt"
        >
          Version
        </TextListItem>
        <TextListItem
          component="dd"
        >
          1.0.0
        </TextListItem>
        <TextListItem
          component="dt"
        >
          Source code
        </TextListItem>
        <TextListItem
          component="dd"
        >
          <a
            href="https://github.com/project-openubl/xml-builder"
          >
            Github
          </a>
        </TextListItem>
        <TextListItem
          component="dt"
        >
          Report issues
        </TextListItem>
        <TextListItem
          component="dd"
        >
          <a
            href="https://github.com/project-openubl/xml-builder/issues"
          >
            Github issues
          </a>
        </TextListItem>
        <TextListItem
          component="dt"
        >
          Documentation
        </TextListItem>
        <TextListItem
          component="dd"
        >
          <a
            href="https://project-openubl.gitbook.io/xml-builder/"
          >
            Documentation
          </a>
        </TextListItem>
        <TextListItem
          component="dt"
        >
          Videos
        </TextListItem>
        <TextListItem
          component="dd"
        >
          <a
            href="https://www.youtube.com/channel/UChq3xxjyDgjcU346rp0bbtA/"
          >
            Youtube
          </a>
        </TextListItem>
        <TextListItem
          component="dt"
        >
          Rest API documentation
        </TextListItem>
        <TextListItem
          component="dd"
        >
          <a
            href="https://app.swaggerhub.com/organizations/project-openubl"
          >
            Swaggerhub
          </a>
        </TextListItem>
        <TextListItem
          component="dt"
        >
          Licence
        </TextListItem>
        <TextListItem
          component="dd"
        >
          <a
            href="https://github.com/project-openubl/xml-builder/blob/master/LICENSE"
          >
            Eclipse Public License - v 2.0
          </a>
        </TextListItem>
      </TextList>
    </TextContent>
  </AboutModal>
</Fragment>
`;
